### PR TITLE
Guide update

### PR DESCRIPTION
**Pull request by reisfmb**
- [x] #48 
- [x] #49 
- [x] #50
- [x] #51
- [x] #56
- [x] #58 
- [x] #60
- [x] #62 
- [x] #64 
- [x] #66 
- [x] #67 
- [x] #68 
- [x] #69 
- [x] #70 
- [x] #71

### CODE REVIEW

File: .github/workflows/enonic-docgen.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - "master"
+      - "guide-update"
```
Comment (poi33) [inline .github/workflows/enonic-docgen.yml:7]: @reisfmb Will we need guide-update in master?

**Review by sigdestad — COMMENTED** [review]
Find comments within the file review section

File: docs/headless-api.adoc [hunk 2]
```diff
@@ -197,51 +133,40 @@ TIP: Learn more about inline fragments at https://graphql.org/learn/queries/#inl
...
 }
 ----
 
+.Getting all data using dataAsJson
+image::query-playground-artists-dataAsJson.png[Query playground query returning artists content together with is name and about data, but using dataAsJson field, {image-xl}]
 
 == Advanced queries
 
-The `query` field underneath `guillotine` allows you to search the system for what you want. This field accepts a number of optional, named arguments to help you hone in on exactly what you're looking for. These arguments are:
+We'll now introduce and show examples of usaged of three different fields underneath `guillotine` field, that allow us to search the system for specific data.
```
Comment (alansemenov) [inline docs/headless-api.adoc:162]:
```suggestion
We'll now introduce and show examples of usages of three different fields underneath the `guillotine` field, that allow us to search the system for specific data.
```

File: docs/headless-api.adoc [hunk 3]
```diff
@@ -197,51 +133,40 @@ TIP: Learn more about inline fragments at https://graphql.org/learn/queries/#inl
...
+
+=== Query (deprecated)
+
+There is a `query` field underneath `guillotine` which allows you to search the system for what you want. This field accepts a number of optional, named arguments to help you find exactly what you're looking for. These arguments are:
 
 `contentTypes`:: The content types we're looking for. Can either be a lone string if you're looking for a single content type, or a list if you want to search among multiple types.
-`filters` :: A list of filters to apply to the results after the query has been executed. We won't get into them here, so if you want to learn more about them, head to the https://developer.enonic.com/docs/xp/stable/storage/filters[filters documentation].
+`filters`:: A list of filters to apply to the results after the query has been executed. We won't get into them here, so if you want to learn more about them, head to the https://developer.enonic.com/docs/xp/stable/storage/filters[filters documentation].
```
Comment (alansemenov) [inline docs/headless-api.adoc:169]:
```suggestion
`filters`:: A list of filters to apply to the results after the query has been executed. We won't get into the details here, so if you want to learn more about them, head to the https://developer.enonic.com/docs/xp/stable/storage/filters[filters documentation].
```

File: .github/workflows/enonic-docgen.yml (hunk 1 above)
Comment (reisfmb) [reply .github/workflows/enonic-docgen.yml:7]: I believe we'll not need that when this is merged :+1:

**Review by sigdestad — COMMENTED** [review]
- [x] Menu is broken for "Sandboxes" - it seems we now have both sandboxex.adoc and sandbox-welcome.adoc

Environment setup

- [ ] Can we simplify the intro by saying what you will do in this step?
- [x] I propose we drop the command line section (it is too basic imho)


Sandboxes

I don't see the new "sandboxes" page in the next branch on dev portal, so the comments are based on the /sandbox-welcome url.

- [ ] Maybe generify username /machine name from command line examples?
- [x] There are two welcome page images

App development

- [ ] The create command creates an app with the wrong name (should be com.example.myproject. I propose we use the following string instead, then you can explan each parameter:  "enonic project create --sb tutorial -r starter-vanilla -d myproject -v 1.0.0 -n com.example.myproject"


Content Studio

- [ ] Assigning an app to a project - shouldn't this happen during the creation of the project? I gues we should go through each wizard step - finishing off with adding the app?
- [ ] "Make sure you’re already running your sandbox with the --dev flag in another terminal, otherwise running enonic project dev will start the sandbox in detached mode, so you’ll lose logs. More on cli’s dev command [here](https://developer.enonic.com/docs/enonic-cli/stable/dev#dev)." - this is wrong afaik? enonic project dev -> start sandbox with --dev

GraphQL

- [ ] Buttons doc is a bit "messy". Would it be possible to show one button at a time, and its explanation? Rather than the grouping and numbering?
- [ ] "fields with connection in their name have new parameters for aggregating data." + pagination
- [ ] "If are not running the project in dev mode.." maybe just make this a "tip" instead?

Medias

- [ ] Still says "headless API" - should say "GraphQL API"
- [ ] "e I’ll add images of some of their albums." I believe the attachment thing can be a bit confusing to fresh devs? How to difference media:images and attachment images? can we maybe add something else in the example? Or, maybe also move attachments stuff into a separate page, after media - or even after input types? Or maybe even to developer 201? Problem is that we have not even talked about references at this point.

Rich text

- [ ] Chapter is still called "headless"


I think thats it.
Great job with the QueryDSL examples btw!

**Comment by reisfmb** [comment]
> "Make sure you’re already running your sandbox with the --dev flag in another terminal, otherwise running enonic project dev will start the sandbox in detached mode, so you’ll lose logs. More on cli’s dev command [here](https://developer.enonic.com/docs/enonic-cli/stable/dev#dev)." - this is wrong afaik? enonic project dev -> start sandbox with --dev

Based on what is said [here](https://developer.enonic.com/docs/enonic-cli/stable/dev#dev), running enonic project dev is not sufficient to get sandbox logs. Read the tip.

**Comment by reisfmb** [comment]
How to proceed with attachments? IMO the explanations are clear, and we can give a `tip` that more will be mentioned on developer 201?

**Comment by reisfmb** [comment]
> Assigning an app to a project - shouldn't this happen during the creation of the project? I gues we should go through each wizard step - finishing off with adding the app?

My idea was to make the assignment of an app to a project a different step to guarantee that the user will be able to do this for further apps... If we just showed on the creation wizard, the assignment procedure would be lost for already existing projects.

**Comment by sigdestad** [comment]
> > "Make sure you’re already running your sandbox with the --dev flag in another terminal, otherwise running enonic project dev will start the sandbox in detached mode, so you’ll lose logs. More on cli’s dev command [here](https://developer.enonic.com/docs/enonic-cli/stable/dev#dev)." - this is wrong afaik? enonic project dev -> start sandbox with --dev
> 
> Based on what is said [here](https://developer.enonic.com/docs/enonic-cli/stable/dev#dev), running `enonic project dev` is not sufficient to get sandbox logs. Read the tip.
You are right about getting the logs. But we can expect sandbox to be running in dev mode already, so lets make this a tip?

**Comment by sigdestad** [comment]
> > "e I’ll add images of some of their albums." I believe the attachment thing can be a bit confusing to fresh devs? How to difference media:images and attachment images? can we maybe add something else in the example? Or, maybe also move attachments stuff into a separate page, after media - or even after input types? Or maybe even to developer 201? Problem is that we have not even talked about references at this point.
> 
> How to proceed with attachments? IMO the explanations are clear, and we can give a `tip` that more will be mentioned on developer 201?

Ok, lets keep it for now.

**Comment by sigdestad** [comment]
> > Assigning an app to a project - shouldn't this happen during the creation of the project? I gues we should go through each wizard step - finishing off with adding the app?
> 
> My idea was to make the assignment of an app to a project a different step to guarantee that the user will be able to do this for further apps... If we just showed on the creation wizard, the assignment procedure would be lost for already existing projects.

But this guide is about setting up your very frist projet, so I believe we should rather explain the wizard, and add the app at the last step.

**Review by alansemenov — APPROVED** [review]

**Pull request merged**